### PR TITLE
Improve test coverage on main

### LINKED ISSUE #60: Improve test coverage on `main.py`
Currently unit tests on `main.py` only reach 70% code coverage. While not a huge problem, as the sections not covered by unit tests are still covered by the end to end tests, it would still be nice to have additional tests to complete coverage on `main.py`, as they could provide insight to issues caused by future changes.

Currently the following methods need additional testing:
 - `do_match`
 - `match_flows`
 - `match_and_apply`

### PR DESCRIPTION
**Pull request by gretelliz**
Closes #60 

### Summary

Add unit tests for:

- do_match
- match_flows
- match_and_apply

Now the coverage is 95% on main.py.

### Local Tests

N/A

### End-to-End Tests

N/A

### CODE REVIEW

File: tests/unit/test_main.py [hunk 1]
```diff
@@ -785,3 +785,132 @@ async def test_get_traces_any(self, mock_stored_flows, event_loop):
         assert result[0][0]["out"] == {"port": 3}
         assert result[1][0]["type"] == "incomplete"
         assert result[1][0]["out"] is None
+
+    def test_do_match(self):
+        """Test do_match."""
+
+        # Check dl_vlan
+        flow = {'flow': {'match': {'dl_vlan': 10}}}
```
Comment (viniarck) [inline tests/unit/test_main.py:793]: The test case looks good, but it's repeating many `args` `resp`, the idiomatic way with `pytest` when you need to test against multiple/different inputs is to use the `@pytest.mark.parametrize` decorator, and then parametrize `flow,args,should_match` (dict, dict, bool), take a look [on some unit tests examples from flow_manager for instance if you need a reference](https://github.com/kytos-ng/flow_manager/blob/master/tests/unit/test_match_13.py#L53-L73)

File: tests/unit/test_main.py [hunk 2]
```diff
@@ -785,3 +785,132 @@ async def test_get_traces_any(self, mock_stored_flows, event_loop):
...
+        stored_flows = {"00:00:00:00:00:00:00:01": [flow]}
+        args = {"dl_vlan": [10], "in_port": 1}
+        resp = self.napp.match_flows(switch, args, stored_flows)
+        assert resp == [flow]
+
+        stored_flows = {}
+        resp = self.napp.match_flows(switch, args, stored_flows)
+        assert not resp
+
+    def test_match_and_apply(self):
```
Comment (viniarck) [inline tests/unit/test_main.py:865]: Same comment regarding `@pytest.mark.parametrize`, ideally it should be using `parametrize` here for maintainability and readability. You can also parametrize the expected `resp` based on what's expected in terms of `flow, entries, port`

**Review by viniarck — CHANGES_REQUESTED** [review]
@gretelliz, it's great to have improved unit test for these methods, in total it went up 1%. Asked a change though regarding how tests were written. 

- Since you're at it, make sure to also nuke `tests/helpers.py` since it's not doing anything other than bringing coverage down, which can also closes https://github.com/kytos-ng/sdntrace_cp/issues/27

Also, in terms of house keeping there are still:

- The module `scheduler.py` and its tests are no longer useful, since scheduling capabilities got removed on https://github.com/kytos-ng/sdntrace_cp/pull/95/. Consequentely, `apscheduler` can also get removed from `requirements/in.txt` and `requirements/run.txt` should be regenerated.
- Readme is also outdated since it's still mentions `amlight/scheduler` as a dependency


```
Name                           Stmts   Miss  Cover
--------------------------------------------------
__init__.py                        0      0   100%
main.py                          169     12    93%
scheduler.py                      18      1    94%
settings.py                        2      0   100%
tests/__init__.py                  6      0   100%
tests/helpers.py                  18     18     0%
tests/unit/__init__.py             0      0   100%
tests/unit/test_main.py          331      0   100%
tests/unit/test_scheduler.py      23      2    91%
tests/unit/test_utils.py         189      3    98%
utils.py                          89      5    94%
--------------------------------------------------
TOTAL                            845     41    95%
```

Once the house keeping tasks are also addressed it's expected that it'll bring the total code coverage up

**Comment by gretelliz** [comment]
@viniarck, total code coverage is 97% now.

**Review by viniarck — APPROVED** [review]
Excellent improved test cov and general house keeping, @gretelliz. 

It's ready to land, but before merging make sure to also update the CHANGELOG mentioning that `apscheduler` is no longer a dependency (in a `Removed` section), since it was a relevant dependency it's important to be highlighted.

**Pull request merged**